projects/cm_mcu/clocksynth.c

```diff
@@ -16,50 +16,51 @@
 #include "Tasks.h"
 
 #ifdef REV2
+#define MAX_TRIES 500
+
 int clear_clk_stickybits(void)
 {
   static_assert(((CLOCK_I2C_BASE == 1) || (CLOCK_I2C_BASE == 2) || (CLOCK_I2C_BASE == 3) ||
                  (CLOCK_I2C_BASE == 4) || (CLOCK_I2C_BASE == 6)),
                 "Invalid I2C base");
+
+  int tries = 0;
   while (getPowerControlState() != POWER_ON) {
     vTaskDelay(pdMS_TO_TICKS(10)); // delay 10 ms
+    ++tries;
+    if (tries > MAX_TRIES)
+      break;
```

Comment:
we should not continue if we fail here, we should return out of the `clear_clk_stickybits` function and signal an error (I guess via the return type). i.e., `return 1;` rather than break or something like that.

Reply:
Done.